### PR TITLE
Create best practices for consumers

### PR DESCRIPTION
**Pull request by aphillips**
w3c/i18n-action#93

Create a section and initial set of BPs for
consumers of lang/dir metadata

***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/string-meta/pull/87.html" title="Last updated on Jul 11, 2024, 3:20 PM UTC (3da573e)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/string-meta/87/af3f6e6...aphillips:3da573e.html" title="Last updated on Jul 11, 2024, 3:20 PM UTC (3da573e)">Diff</a>

### CODE REVIEW

**Comment by xfq** [comment]
We should probably add some text/links about how to normalize language tags to help ensure interoperability.

File: index.html [hunk 1]
```diff
@@ -507,22 +507,42 @@ <h3 id="bp-producers">Guidance for [=Producers=]</h3>
...
-<p class="advisement">Consumers SHOULD consume direction metadata provided by document formats or protocols whenever possible. When direction metadata is not available, consumers SHOULD use first-strong detection (possibly by using the value <code>auto</code> for the direction).</p>
+<p class="advisement">Consumers SHOULD employ any [=string direction=] metadata provided by document formats or protocols when processing or displaying the associated string value.</p>
 </div>
 
 <div class="req" id="bp-consumer-isolation">
-<p class="advisement">When inserting content into a document, and particularly when inserting that content into existing text, consumers SHOULD apply bidirectional isolation via markup or Unicode isolating characters. Such isolation SHOULD use the direction found in document format or protocol supplied metadata if possible.</p>
+<p class="advisement">When displaying a string or inserting a string into a document, consumers SHOULD apply bidirectional isolation to the string value using the [=string direction=] found in document format or protocol metadata when available or first-strong isolation when it is not.</p>
```
Comment (r12a) [inline index.html]: This is a very long, meandering sentence, that packs in a lot of information, but in a way that is not very easy to read.  I suggest replacing this mustard with two mustards as follows:

````
<div class="req" id="bp-consumer-isolation">
<p class="advisement">When a string is displayed in or inserted into a document, consumers SHOULD isolate it directionally from the surrounding text.</p>
</div>`

<div class="req" id="bp-consumer-direction">
<p class="advisement">Consumers SHOULD apply direction to a string when it is inserted into a document. If [=string direction=] is provided by metadata associated with a string, consumers SHOULD use that. If string metadata is not available, first-strong heuristics should be used to assign a direction.</p>
</div>
````

File: index.html [hunk 2]
```diff
@@ -507,22 +507,42 @@ <h3 id="bp-producers">Guidance for [=Producers=]</h3>
...
+</div>
+
+<p>To prevent [=spillover effects=] and get the best results, it is always safe to wrap an inserted string value with bidirectional isolation.</p>
```
Comment (r12a) [inline index.html]: Suggestion:

```
<p>It never causes a problem to wrap an inserted string value with bidirectional isolation, but it prevents [=spillover effects=] and produces the best results.</p>
```

File: index.html [hunk 3]
```diff
@@ -507,22 +507,42 @@ <h3 id="bp-producers">Guidance for [=Producers=]</h3>
...
+
+<p>In an [[HTML]] document, for example, this means placing a <code>dir</code> attribute on an element directly around the inserted string value. If the string is inserted inside another string, you can use an inline element such as <code>span</code>, <code>em</code>, or <code>bdi</code> to carry the <code>dir</code> attribute.</p>
```
Comment (r12a) [inline index.html]: There are some small innacuracies here.  I suggest the following alternative text:

```
<p>In an [[HTML]] document, for example, this means placing a <code>dir</code> attribute on an existing element that directly wraps the inserted string value. If no existing markup is available, you can use an inline element such as <code>span</code>, or <code>bdi</code> to carry the <code>dir</code> attribute.</p>
```

The above assumes that you know the direction to apply, either through interpreting metadata, or by first-strong heuristics.  If those don't apply, one should use the bdi element without a dir attribute.  Should we make that assumption? Or should we add a paragraph that says what i just wrote above?

**Review by r12a — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (aphillips) [reply index.html]: @r12a If you're going to go to the trouble of typing you suggestion out, consider using \`\`\`suggestion to mark up the suggested text. Like so:

```suggestion
<div class="req" id="bp-consumer-isolation">
<p class="advisement">When a string is displayed in or inserted into a document, consumers SHOULD isolate it directionally from the surrounding text.</p>
</div>`

<div class="req" id="bp-consumer-direction">
<p class="advisement">Consumers SHOULD apply direction to a string when it is inserted into a document. If [=string direction=] is provided by metadata associated with a string, consumers SHOULD use that. If string metadata is not available, first-strong heuristics SHOULD be used to assign a direction.</p>
</div>
```

File: index.html (hunk 2 above)
Comment (aphillips) [reply index.html]: @r12a suggested

```suggestion
<p>It never causes a problem to wrap an inserted string value with bidirectional isolation, but it prevents [=spillover effects=] and produces the best results.</p>
```

Comment (aphillips) [reply index.html]: Here's my light edit:

```suggestion
<p>It never causes a problem to wrap an inserted string value with bidirectional isolation, and doing so prevents [=spillover effects=] to produce the best result.</p>
```

File: index.html (hunk 3 above)
Comment (aphillips) [reply index.html]: @r12a suggested:

```suggestion
<p>In an [[HTML]] document, for example, this means placing a <code>dir</code> attribute on an existing element that directly wraps the inserted string value. If no existing markup is available, you can use an inline element such as <code>span</code>, or <code>bdi</code> to carry the <code>dir</code> attribute.</p>
```

**Pull request merged**